core/CronArchive.php

```diff
@@ -1384,6 +1398,52 @@ private function makeWebsiteIdArchiveList(array $websitesIds)
         return new SharedSiteIds($websitesIds, SharedSiteIds::OPTION_ALL_WEBSITES);
     }
 
+    private function getPluginNameForArchiveIfAny($archive)
+    {
+        $name = $archive['name'];
+        if (strpos($name, '.') === false) {
+            return null;
+        }
+
+        $parts = explode('.', $name);
+        return $parts[1];
+    }
+
+    private function getArchivingAPIMethod($plugin)
+    {
+        $cache = Cache::getTransientCache();
+        $cacheKey = 'CronArchive.getArchivingAPIMethodForPlugin.' . $plugin;
+
+        $method = $cache->fetch($cacheKey);
+        if (!empty($method)) {
+            return $method;
+        }
+
+        $method = "$plugin.get";
+
+        /**
+         * By default the core:archive command will use `YourPlugin.get` to initiate archiving for
+         * a plugin when archiving for just one plugin.
+         *
+         * If your plugin has an archiver but does not provide a `get` API method for general metrics,
+         * then you can use this method to specify which API method to use to make sure the plugin data
+         * is archived.
+         *
+         * @param string &$method Set this value to the method if `$plugin` is your plugin.
+         * @param string $plugin The name of the plugin.
+         */
+        Piwik::postEvent('CronArchive.getArchivingAPIMethodForPlugin', [&$method, $plugin]);
```

Comment:
Be great if this maybe wasn't needed like if there was an API like `CoreAdminHome.archivePluginSpecificReport($plugin, $requestedReport, $idSite, $period, $date, $segment = false)` which then would maybe directly bypass the archiver and this API method could instantiate the requested archiver directly etc. This would make it easier for plugin developers as such a method wouldn't need to be specified plus some logic would maybe get easier? Not sure... might be also faster though. Guess it would do something like this: https://github.com/innocraft/plugin-CustomReports/blob/4.x-dev/Commands/ArchiveReports.php#L59-L226

Not replacing entries within an existing archive though but adding a new DONE PARTIAL archive.

The advantage could be that this method would be also callable separately for any plugin maybe. The method would also check if `SettingsServer::isArchivePhpTriggered() = true`.

Not sure it's that easy though or if it in the end makes the logic a lot easier. The reading of archives be still as "complicated".

Reply:
Could make it more complicated I guess as it would still need to fetch all archiveIds and check if reprocessing is needed etc and would probably add some duplicated code there maybe. So feel free to ignore it unless you want to explore it.

Just thought be nice not needing this event.

Reply:
Well if we're archiving for a specific report, we can just try to call that API method. However, if it's the entire plugin archive that's invalidated, then we don't necessarily know which API method to call.

I guess we could just not do API.get and do just a `API.initiateArchiving` method, but then there would be two code paths for initiating archiving, one through Archive.php and one through this other API method.